libraries/core_libs/network/src/tarcap/packets_handlers/pbft_sync_packet_handler.cpp

```diff
@@ -99,7 +109,7 @@ void PbftSyncPacketHandler::process(const PacketData &packet_data, const std::sh
   }
 
   // Check cert vote matches
-  for (auto const &vote : sync_block.cert_votes) {
+  for (auto const &vote : cert_votes) {
```

Comment:
so we are checking this only for the latest block, is that ok?

Reply:
What does it mean we did not get it yet ? Yo can (and probably should ? ) perform this check for both  sync_block.cert_votes as well as cert_votes

Reply:
I meant that we could not verify the current block unless it is the last one before we get the votes that are in the next block. There is a full vote verification when we try to push blocks into chain but I added an additional check in PbftManager::PeriodDataQueuePush so that we do not push in the queue if the votes are invalid

Reply:
wait, this field `sync_block.cert_votes` are basically cert_votes for the previous block if I understand it correctly ? So you can definitely verify correctness of these votes by checking if those are the votes for the previous block + you can verify if the cert_votes for the last block are pointing to it

Reply:
Yes and I am doing this in PbftManager::PeriodDataQueuePush